### PR TITLE
netatalk 3.1.13 -> 3.1.13_3

### PR DESCRIPTION
**Pull request by jcumming**
###### Description of changes

Pull in FreeBSD patches from:
  https://bugs.freebsd.org/263123
  https://bugs.freebsd.org/251203

Thanks to Joe Marcus Clarke <marcus@FreeBSD.org> for putting this together.

This fixes crashes from the 3.1.13 update, and addresses https://github.com/NixOS/nixpkgs/issues/189510

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by onny** [comment]
Please squash commits

**Comment by jcumming** [comment]
> Please squash commits

Please be helpful and encourage contributors? 

is there a way to do that on existing pull requests, or do I have to rebase and squash and make a new pull request?

**Comment by pshirshov** [comment]
In fact there is a convenient repository setting on github allowing to do squashes when someone accepts P/Rs...

Don't know why it's not on and so many people are bugged by these "squash commits" comments. Just in case, this is a fix for a critical bug.

Screenshot_20221013_151938

**Comment by trofi** [comment]
Result of `nixpkgs-review pr 193550` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>netatalk</li>
  </ul>
</details>

**Comment by trofi** [comment]
> > Please squash commits
> 
> Please be helpful and encourage contributors?
> 
> is there a way to do that on existing pull requests, or do I have to rebase and squash and make a new pull request?

Existing pull requests are completely fine to modify. You can squash commits by checking out the branch locally and use `git rebase -i origin/master` (`squash` operation instead of the `pick` one)

File: pkgs/tools/filesystems/netatalk/default.nix [hunk 1]
```diff
@@ -22,6 +24,68 @@ stdenv.mkDerivation rec {
     })
   ];
 
+  freeBSDPatches = [
+    # https://bugs.freebsd.org/263123
+    (fetchpatch {
+      name = "patch-etc_afpd_directory.c";
+      url = "https://cgit.freebsd.org/ports/plain/net/netatalk3/files/patch-etc_afpd_directory.c";
+      sha256 = "03zgijjaxpqdsrhavmsingx7i8a5v5869nqxrxyczkzknr20fw55";
```
Comment (trofi) [inline pkgs/tools/filesystems/netatalk/default.nix:32]: The hashes seem to fail the fetch (posted as a build review failure).

**Review by trofi — CHANGES_REQUESTED** [review]

**Comment by pshirshov** [comment]
@trofi : seems like the build passes, could you merge?..

**Comment by trofi** [comment]
> @trofi : seems like the build passes, could you merge?..

I tried again to build it as:

```
$ nix build github:NixOS/nixpkgs/pull/193550/merge#netatalk
error: hash mismatch in fixed-output derivation '/nix/store/43f8dd2dcr4xsxflryqag4nli0livslw-patch-etc_cnid__dbd_cmd__dbd__scanvol.c.drv':
         specified: sha256-1M7YeYEZThsV/XZdCEL2DnZSrTEvY/1aC70lsaEDEE0=
            got:    sha256-5QV+tQDo8/XeKwH/e5+Ne+kEOl2uvRDbHMaWysIB6YU=
error: 1 dependencies of derivation '/nix/store/wl6yby1p0jcm7mj4vlgxwvh6ng34ddci-netatalk-3.1.13_3.drv' failed to build
```

I think it's still the same issues I see in https://github.com/NixOS/nixpkgs/pull/193550#pullrequestreview-1141097836. Does the command work for you as is?

**Comment by jcumming-solidigm** [comment]
Working on cleaning this up just now..

**Pull request closed without merging**